mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java

```diff
@@ -166,6 +166,24 @@ private Mailbox addAcl(MailboxACL acl, Mailbox mailbox) {
                 () -> LOGGER.warn("Could not retrieve mailbox {} with path {} in mailbox table.", idAndPath.getCassandraId(), idAndPath.getMailboxPath())));
     }
 
+    @Override
+    public MailboxId create(Mailbox mailbox) throws MailboxException {
+        CassandraId cassandraId = retrieveId(mailbox);
+        mailbox.setMailboxId(cassandraId);
+        if (!tryCreate(mailbox, cassandraId)) {
+            throw new MailboxExistsException(mailbox.generateAssociatedPath().asString());
+        }
+        return cassandraId;
+    }
+
+    private boolean tryCreate(Mailbox cassandraMailbox, CassandraId cassandraId) {
+        boolean isCreated = mailboxPathV2DAO.save(cassandraMailbox.generateAssociatedPath(), cassandraId).block();
```

Comment:
could you explain why the `trySave` we call `mailboxPathV2DAO.delete` ? I read the JIRA and still can't understand the logic and why we need a new method to avoid doing this call.

Reply:
Sure. save is combining two concerns: create and rename. Upon a rename you need to 'clear' the previous path.

This is unecessary upon 'create' and this extra operation can lead to increased inconsistency exposure.

So along with a better code organisation separating 'create' from 'rename' will allow to have a better robustness as well.

Reply:
Sure !
Let's say I have a mailbox with :
* Id : 1
* Name (or path) : mailbox1
I want to rename it to mailbox2.
 
In the `trySave`, what is done is this:
* we save the new `mailboxPath` mailbox2 with the ID 1 (`mailboxPathV2DAO.save`)
* we fetch the current mailbox we want to rename with `retrieveMailbox`, so mailbox(id: 1, path: mailbox1)
* we delete the old path that we get from the mailbox we just fetched (mailbox1) -> `mailboxPathV2DAO.delete`
* we save the new mailbox (id:1, path: mailbox2)

That `retrieveMailbox` then it's only useful in the case of a rename mailbox scenario. If you create one, it's unecessary (and you don't call the `mailboxPathV2DAO.delete` as well). We thought it's better to separate that logic as it increases the chances of failure upon creation. If the read fails, then the mailbox is not saved, we have an inconsistency.

I think we could raise the point as well that the old mailbox entry is probably still there? (without the one in mailboxPath). I don't know if this was intended or not when this was implemented, but we can discuss about it, but this is not the point of this PR.

Reply:
Thank you very much for the explanation